src/utils/config.py

```diff
@@ -43,49 +45,22 @@ class SearchModels():
 
 
 class TextSearchConfig():
```

Comment:
Would making the `TextSearchConfig` directly inherit from `ConfigBaseModel` simplify the code even further? That is have `TextSearchConfig(ConfigBaseModel)`